addons/base/views.py

```diff
@@ -274,8 +274,34 @@ def get_auth(auth, **kwargs):
         raise HTTPError(httplib.BAD_REQUEST)
 
     try:
-        credentials = provider_settings.serialize_waterbutler_credentials()
-        waterbutler_settings = provider_settings.serialize_waterbutler_settings()
+        path = data.get('path')
+        version = data.get('version')
+        credentials = None
+        waterbutler_settings = None
+        if provider_name == 'osfstorage':
+            if path and version:
+                fileversion = FileVersion.objects.get(basefilenode___id=path.strip('/'), identifier=version)
```

Comment:
Hmm, I think since it's an expected behavior for version to not be passed at all... so would it make sense if the version is incorrect it defaults to just using the region for NodeSettings just as it does if there is no version included? I can see going either way, just defaulting to the node's nodesettings region, or erroring because it's clearly asking for something that's not there...

Reply:
Yeah, could go either way. Probably better to be strict in this case. If WB is passing a version that doesn't exist, we're doing something wrong and should be notified about it.

Reply:
The `version` WB passes is based entirely on user input.  It might be present, it might be absent, or it might be wrong.  If it's absent, the assumption would be that the most recent version is desired.  If it's wrong, then a 404 error would be appropriate.

Reply:
Ah, thanks @felliott .